src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -2687,6 +2687,49 @@ LogicalResult ONNXConstantOp::inferShapes(
 // Concat
 //===----------------------------------------------------------------------===//
 
+static LogicalResult verify(ONNXConcatOp op){
+  ONNXConcatOpAdaptor operandAdaptor =
+    ONNXConcatOpAdaptor(op);
+
+  int inputNum = op.getNumOperands();
+  // Check all inputs.
+  for (int i = 0; i < inputNum; ++i) {
+    if (!hasShapeAndRank(operandAdaptor.getOperands()[i])) {
+      // Won't be able to do any checking at this stage.
+      return success();
+    }
+  }
+  // Checking value of axis parameter.
+  auto commonType = operandAdaptor.getOperands()[0].getType().cast<RankedTensorType>();
+  auto commonShape = commonType.getShape();
+  int64_t commonRank = commonShape.size();
+  int64_t axisIndex = op.axis();
+  if (axisIndex < 0)
+    axisIndex = commonRank + axisIndex;
+  if (axisIndex >= commonRank)
+    return op->emitError("Concat axis value out of bound");
+
+  for (int i = 1; i < inputNum; ++i) {
+    auto currShape =
+        operandAdaptor.getOperands()[i].getType().cast<RankedTensorType>().getShape();
+    if ((int64_t)currShape.size() != commonRank)
+      return op->emitError("Concat input must all have the same rank");
+    for (int j = 0; j < commonRank; ++j) {
+      if (j == axisIndex) {
```

Comment:
Suggest to use ` continue` here rather than an empty if block:
```
if (j == axisIndex) 
  continue;
  
if (currShape[j] != -1 && commonShape[j] != -1 &&
                 currShape[j] != commonShape[j]) {
        return op->emitError(...);
```

In general consider using early-exits in loops, they make the code more readable.   
One more nit :-)  Avoid using `{}` for single statements (https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements)